File: src/AdventureWorks.Web.Api/ClientApp/src/app/models/data/entities/sales/shopping-cart-item/shopping-cart-item-lookup-model.ts
import { IProductLookupModel } from 'app/models/data/entities/production/product/product-lookup-model';

export interface IShoppingCartItemLookupModel {
  shoppingCartItemID: number;
  shoppingCartID: string;
  quantity: number;
  productID: number;
  dateCreated: Date | string;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */
}

export class ShoppingCartItemLookupModel implements IShoppingCartItemLookupModel {
  shoppingCartItemID: number;
  shoppingCartID: string;
  quantity: number;
  productID: number;
  dateCreated: Date | string;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */

  constructor(init?: Partial<ShoppingCartItemLookupModel>) {
    if (!!init) {
      Object.assign<ShoppingCartItemLookupModel, Partial<ShoppingCartItemLookupModel>>(this, init);
    }
  }
}

File: src/AdventureWorks.Application/DataEngine/Sales/ShoppingCartItem/Queries/GetShoppingCartItems/ShoppingCartItemLookupModel.cs
using System;
using AutoMapper;
using AdventureWorks.Application.Interfaces.Mapping;
using AdventureWorks.BaseDomain.Entities.Sales;
using System.Collections.Generic;
using AdventureWorks.Common.Extensions;
using AdventureWorks.BaseDomain.Enums;
using AdventureWorks.BaseDomain.Interfaces;
using AdventureWorks.BaseDomain.CustomTypes;
using AdventureWorks.Application.DataEngine.Production.Product.Queries.GetProducts;

namespace AdventureWorks.Application.DataEngine.Sales.ShoppingCartItem.Queries.GetShoppingCartItems
{
    public partial class ShoppingCartItemLookupModel : IHaveCustomMapping
    {
        public int ShoppingCartItemID { get; set; }
        public string ShoppingCartID { get; set; }
        public int Quantity { get; set; }
        public int ProductID { get; set; }
        public DateTime DateCreated { get; set; }
        public DateTime ModifiedDate { get; set; }

        #region Navigation Properties

        // Uncomment the following line if you need child navigation properties (can cause a poor performance)
        /*  */
        // Uncomment the following line if you need parent navigation properties
        /*  */
        #endregion

        public void CreateMappings(Profile configuration)
        {
            configuration.CreateMap<Domain.Entities.Sales.ShoppingCartItem, ShoppingCartItemLookupModel>().IgnoreMissingDestinationMembers().CompensateWithDestinationValues();
            configuration.CreateMap<ShoppingCartItemLookupModel, BaseShoppingCartItem>().IgnoreMissingDestinationMembers();
        }
    }
}
